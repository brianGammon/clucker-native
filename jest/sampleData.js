const sampleData1 = {
  userSettings: {
    key: 'user1',
    data: {
      currentFlockId: 'flock1',
    },
  },
  flocks: {
    data: {
      flock1: {
        name: 'Test Flock 1',
        ownedBy: 'user1',
      },
      flock2: {
        name: 'Test Flock 2',
        ownedBy: 'user2',
      },
    },
  },
  chickens: {
    data: {
      chicken1: {
        name: 'The First Chicken',
      },
      chicken2: {
        name: 'The Second Chicken',
      },
    },
  },
  eggs: {
    data: {
      egg1: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-01',
        modified: '2018-01-12T18:05:54.806Z',
        notes:
          'This was the record setting egg from May 2017, real egg was 66.1g on this day',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg2: {
        chickenId: 'chicken2',
        chickenName: 'Baby',
        damaged: false,
        date: '2017-09-24',
        modified: '2017-10-12T18:05:54.806Z',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg3: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-02',
        modified: '2018-08-02T18:05:54.806Z',
        notes: '',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 68.1,
      },
    },
  },
};

export default sampleData1;
