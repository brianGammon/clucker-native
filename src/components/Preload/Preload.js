/* @flow */
import * as React from 'react';
import { AsyncStorage } from 'react-native';
import { connect } from 'react-redux';
import Loading from '../Loading';
import { type User } from '../../types';

type Props = {
  navigation: any,
  user: User,
};

class Preload extends React.Component<Props> {
  async componentDidMount() {
    const { navigation, user } = this.props;
    const hasSignedIn = await AsyncStorage.getItem('hasSignedIn');
    const SignedOutRoute = hasSignedIn ? 'SignIn' : 'SignUp';
    navigation.navigate(user ? 'SignedIn' : SignedOutRoute);
  }

  render() {
    return <Loading message="Checking Status..." />;
  }
}

const mapStateToProps = ({ auth: { user } }) => ({
  user,
});

export default connect(mapStateToProps)(Preload);
