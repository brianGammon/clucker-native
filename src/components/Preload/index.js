import PreloadComponent from './Preload';

export default PreloadComponent;
