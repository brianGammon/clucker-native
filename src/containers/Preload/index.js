import React from 'react';
import {
  AsyncStorage, ActivityIndicator, StatusBar, View,
} from 'react-native';
import firebase from 'react-native-firebase';
import styles from './styles';

export interface Props {
  navigation: any;
}

export interface State {}

export default class Preload extends React.Component<Props, State> {
  // Fetch the token from storage then navigate to our appropriate place
  async componentDidMount() {
    const { navigation } = this.props;

    firebase.auth().onAuthStateChanged((user) => {
      console.log(user);
    });

    const userToken = await AsyncStorage.getItem('userToken');

    // This will switch to the App screen or Auth screen and this loading
    // screen will be unmounted and thrown away.
    navigation.navigate(userToken ? 'SignedIn' : 'SignedOut');
  }

  // Render any loading content that you like here
  render() {
    return (
      <View style={styles.container}>
        <ActivityIndicator size="large" />
        <StatusBar barStyle="default" />
      </View>
    );
  }
}
