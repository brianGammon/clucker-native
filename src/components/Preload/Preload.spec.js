import React from 'react';
import configureStore from 'redux-mock-store';
import sampleData from '../../../jest/test-data/full.2017-10-to-2018-05.json';
import Preload from '.';

describe('Preload component:', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const store = mockStore(sampleData);

  test('Should render', () => {
    const navigation = {
      navigate: () => {},
    };
    const wrapper = shallow(<Preload navigation={navigation} store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
