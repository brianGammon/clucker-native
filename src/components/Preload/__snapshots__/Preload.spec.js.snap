// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preload component: Should render 1`] = `
<Preload
  dispatch={[Function]}
  navigation={
    Object {
      "navigate": [Function],
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={
    Object {
      "email": "test@example.com",
      "uid": "user1",
    }
  }
/>
`;
